js/assign/assign.js

```diff
@@ -22,9 +22,20 @@
  *
  * @return {Object} Returns the `target` argument.
  */
-module.exports = function (d, s) {
-	for (var prop in s) {
-		d[prop] = s[prop];
-	}
-	return d;
-};
+ module.exports = function (target, args) {
+ 	if (target === null) {
+   	throw new TypeError('The target object cannot be undefined or null');
```

Comment:
do we need this type check?  Maybe dev-mode only?   Btw === null won't check undefined.